metrics/tags.go

```diff
@@ -5,8 +5,83 @@ import (
 	"encoding/json"
 	"sort"
 	"strings"
+
+	"github.com/mstoykov/atlas"
 )
 
+// TagSet represents a set of tags.
+type TagSet struct {
+	tags *atlas.Node
+}
+
+// NewTagSet creates a TagSet, if a not-nil map is passed
+// then it adds all the pairs to the set, otherwise an empty set is created.
+// Under the hood it initializes an Atlas root node.
+// It should be only used in a centralized place and
+// all the new TagSet should branch from it for getting
+// the optimal performances.
+func NewTagSet(m map[string]string) *TagSet {
```

Comment:
Connected to my other comment (https://github.com/grafana/k6/pull/2594/files#r936302703), I kind of think this function shouldn't exist at all, or if it does exist, it should definitely be unexported and used privately only in a few places in the metrics package :thinking: 

Otherwise, if it exists in its current state, it's going to be misused, since people might not realize that comparing tag sets will fail if they don't share the same root. Even it's current comment is a bit misleading, it says "all the new TagSet should branch from it for getting the optimal performances." when it should probably say "all the new TagSets should branch from it or everything will be broken" :sweat_smile: